isLike 기능 구현

Nitoriubar opened the pull request:
## 📝 세부사항
로그인 한 사용자가 전체 게시물을 불러올 경우 사용자가 좋아요한 유무에 따라 단일 게시물과 좋아요 유무( Boolean값)를 생성자의 파라미터로 넘겨 `GetPostResponse`생성하고 매핑해주는 기능 구현함

## 📗 참고사항

## 📎 관련 이슈
#23 

close #

Nitoriubar committed:
feat : isLike 기능 구현완료

sangrimlee commented on src/main/java/com/triprint/backend/domain/post/controller/PostController.java:
```diff
@@ -47,7 +47,7 @@ public class PostController {
 		@PageableDefault(sort = "id", direction = Direction.DESC) Pageable page,
 		@CurrentUser UserPrincipal userPrincipal) {
-		return ResponseEntity.ok(postService.getPostList(page)); //로그인 X => null
+		return ResponseEntity.ok(postService.getPostList(page, userPrincipal.getId())); //로그인 X => null
```
로그인하지 않은 경우에는 `userPrincipal`이 `null`인 건가요? 그렇다면, `userPrincipal.getId()`를 실행할 때 에러가 발생하지 않을까요?

Nitoriubar replied:
수정 완료했습니다! 감사합니다 :)

sangrimlee commented on src/main/java/com/triprint/backend/domain/post/service/PostService.java:
```diff
@@ -82,12 +92,13 @@ public PostResponse create(Long authorId, CreatePostRequest createPostRequest, L
 			throw new ResourceNotFoundException("해당하는 게시물이 존재하지 않습니다.");
 		});
-		return new GetPostResponse(post);
+		return new GetPostResponse(post, true);
```
단일 포스트 가져오는 것에서도 사용자의 좋아요 여부를 판별해야 할 것 같아요!

Nitoriubar replied:
넵! 수정 완료했습니다!

sangrimlee reviewed (approved):
고생하셨습니다👍

Nitoriubar committed:
fix : isLike feedback 수정 완료

- controller에서 user가 null인 경우 에러 발생하므로 -1L로 변경하여 넘김
- 단일 게시물에서도 List와 같이 구현해야하므로 수정함

Nitoriubar committed:
fix : 로그인 유무 관련하여 게시물의 좋아요 유무 수정
